### PR TITLE
Transfer job sandbox to dcache

### PR DESCRIPTION
**Pull request by retzkek**
This is the first part of addressing #147. It modifies the wrapper script to transfer the truncated output (stdout and stderr) to dCache, as well as the job executable and jdf from the process 0. 

New/modified template variables:
* `date` - changed to be only the date: `%Y_%m_%d`
* `datetime` - what was previously `date`, including the time: `%Y_%m_%d_%H%M%S`
* `outurl` - base URL for the job sandbox in dCache, which is  `$JOBSUB_OUTPUT_URL/{{date}}/{{uuid}}`

New/modified classad attributes;
* `JobsubOutputURL`
* `JobsubUUID`

New  environment variable `JOBSUB_OUTPUT_URL`; if not set submission will continue without the transfers being included. A separate PR will be made with the config change.

~TODO: only simple templates are currently changed, still need to add this to dag templates.~

### CODE REVIEW

**Comment by retzkek** [comment]
I believe this is ready, would appreciate a review. I've rebased onto the latest master.

File: templates/dataset_dag/dagbegin.cmd [hunk 1]
```diff
@@ -4,7 +4,7 @@ universe           = vanilla
 executable         = sambegin.sh
 arguments          =
 
-{% set filebase %}sambegin.$(Cluster)$(Process){% endset %}
+{% set filebase %}sambegin.$(Cluster).$(Process){% endset %}
```
Comment (shreyb) [inline templates/dataset_dag/dagbegin.cmd:7]: Thanks for fixing this on the way.

File: templates/dataset_dag/sambegin.sh [hunk 2]
```diff
@@ -50,16 +91,17 @@ chmod +x ${JSB_TMP}/ifdh.sh
...
 #EXPERIMENT=$1
 #DEFN=$2
 #PRJ_NAME=$3
 #GRID_USER=$4
 
 export JSB_TMP=$_CONDOR_SCRATCH_DIR/jsb_tmp
 mkdir -p $JSB_TMP
+redirect_output_start
 setup_ifdh_env
+echo `date` BEGIN executing /fife/local/scratch/uploads/fermilab/mengel/2020-01-14_162931.318256_6730/fife_wrap_20200114_162957_3316054.sambegin.sh
```
Comment (shreyb) [inline templates/dataset_dag/sambegin.sh:103]: This isn't just for you, Kevin, but also for @marcmengel .  Why are we saying here that we're starting a script that's in Marc's area...on /fife/local?  This seems very wrong...and possibly a holdover from testing.  Please correct me if I'm wrong.

File: templates/dataset_dag/samend.sh [hunk 3]
```diff
@@ -48,11 +89,13 @@ chmod +x ${JSB_TMP}/ifdh.sh
 }
 
 
-echo `date` BEGIN executing /fife/local/scratch/uploads/fermilab/mengel/2020-01-14_162931.318256_6730/fife_wrap_20200114_162957_3316054.samend.sh
->&2 echo `date` BEGIN executing /fife/local/scratch/uploads/fermilab/mengel/2020-01-14_162931.318256_6730/fife_wrap_20200114_162957_3316054.samend.sh
 export JSB_TMP=$_CONDOR_SCRATCH_DIR/jsb_tmp
 mkdir -p $JSB_TMP
+redirect_output_start
 setup_ifdh_env
+echo `date` BEGIN executing /fife/local/scratch/uploads/fermilab/mengel/2020-01-14_162931.318256_6730/fife_wrap_20200114_162957_3316054.samend.sh
```
Comment (shreyb) [inline templates/dataset_dag/samend.sh:96]: Same comment as sam_begin.sh

**Review by shreyb — CHANGES_REQUESTED** [review]
This largely looks good.  I'm concerned about the echo statements talking about /fife/local scripts being executed.

File: templates/dataset_dag/sambegin.sh (hunk 2 above)
Comment (retzkek) [reply templates/dataset_dag/sambegin.sh:103]: Well that's embarrassing that I moved those lines but didn't notice that, good catch. Probably a "generated" script got mixed up with the template at some point. We could probably just change it to "BEGIN executing sambegin.sh"?

Comment (shreyb) [reply templates/dataset_dag/sambegin.sh:103]: That sounds good.

Comment (retzkek) [reply templates/dataset_dag/sambegin.sh]: changed

File: templates/dataset_dag/samend.sh (hunk 3 above)
Comment (retzkek) [reply templates/dataset_dag/samend.sh]: changed

**Review by shreyb — COMMENTED** [review]
LGTM.  Not merging, since we're cutting a release in 24 hours.  I'll approve and merge after that.

**Review by shreyb — DISMISSED** [review]

**Pull request merged**